### PR TITLE
feat(): additional circle ci parameters

### PR DESCRIPTION
**Pull request by danielciuraru86**
# Description

[JIRA-11818](https://scalock.atlassian.net/browse/SAAS-11818)

Support for circle-ci in billy

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Scope of change

- [ ] Customer Assets will get updated
- [ ] All Repositories will be queued for offline scans
- [x] Changes in UI

## Does this need changes in CloudDeployments?

- [ ] new envs
- [ ] new secrets

## Needs a migration script?

_link to the script_

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Unit Tests
- [x] Integration Tests
- [x] Manual Tests - Local Environment
- [x] Manual Tests - Dev Environment
- [ ] Automatic Tests - Dev Environment

# Screenshots / Video
image

image

# Links to related PRs
https://github.com/argonsecurity/billy/pull/294
https://github.com/argonsecurity/go-environments/pull/27

### CODE REVIEW

File: environments/circleci/environment.go [hunk 1]
```diff
@@ -42,7 +47,7 @@ var (
 type environment struct{}
 
 func (e environment) GetBuildLink() string {
-	return ""
+	return os.Getenv(buildUrlEnv)
```
Comment (tamirkiviti13) [inline environments/circleci/environment.go:50]: If it is possible, please provide also a link to the step

File: environments/circleci/environment_test.go [hunk 2]
```diff
@@ -59,9 +71,13 @@ func Test_environment_GetConfiguration(t *testing.T) {
...
 					Id: "49",
+					SourceRef: models.Ref{
+						Branch: "main",
+					},
 				},
-				Environment: enums.CircleCi,
-				ScmId:       "8891c0db39f3064732cc1b4ac02c9b9f",
+				Environment:   enums.CircleCi,
+				ScmId:         "8891c0db39f3064732cc1b4ac02c9b9f",
+				PipelinePaths: []string{""},
```
Comment (tamirkiviti13) [inline environments/circleci/environment_test.go:80]: Please add in the test data a mock repo with a mock pipeline

File: environments/circleci/environment.go (hunk 1 above)
Comment (danielciuraru86) [reply environments/circleci/environment.go:50]: CircleCi doesn't seem to give an env for this:
https://dev.azure.com/danielciuraru/bugbashing/_git/empty_repo

File: environments/circleci/environment_test.go [hunk 3]
```diff
@@ -44,6 +44,18 @@ func Test_environment_GetConfiguration(t *testing.T) {
...
 					CloneUrl: testRepoCloneUrl,
 					Source:   enums.Github,
+					Url:      testRepoUrl,
+				},
+				Pipeline: models.Pipeline{
+					Entity: models.Entity{
+						Id:   "test_workflow",
+						Name: "test_workflow",
+					},
+					Path: "/tmp/circleci/repo/.circleci/config.yml",
```
Comment (tamirkiviti13) [inline environments/circleci/environment_test.go:54]: `fmt.Sprintf("%s/%s", testRepoPath, pipelinePath)`

**Review by tamirkiviti13 — APPROVED** [review]

**Review by orizerah — APPROVED** [review]

**Pull request merged**